### ISSUE TITLE
[BUG] pip installation results in missing functionality on Google Colab

### ISSUE THREAD

**Issue by sokol11** [rapidsai/cudf#12460]
Hi. Pip installation on Google Colab with `!pip install cudf-cu11 --extra-index-url=https://pypi.ngc.nvidia.com` seems to work but results in missing functionality. For example, trying to compute column correlations in a dataframe with a `min_periods` argument specified raises `NotImplementedError: Unsupported argument 'min_periods'`. Other general functionality seems to be missing as well with various errors raised. Interestingly, everything seemed to be working yesterday. Any thoughts on what could be going on? Having a working pip installation on colab would be a game changer!

**Comment by beckernick** [comment]
The 22.12 pip packages provide cuDF as of version 22.12. The `DataFrame.corr` option `min_periods` is [not yet supported,](https://github.com/rapidsai/cudf/blob/8a1ff584e39084b588b82a4d8a5f909bd7451916/python/cudf/cudf/core/dataframe.py#L6495-L6496), which is why you're see a NotImplementedError. I'm going to convert this issue into a feature request for `min_periods`.

Would you be able to file separate feature requests for other functionality that you tried to use but an into errors?

**Comment by sokol11** [comment]
@beckernick yes, thank you and appreciate it! One thing I found confusing is the documentation here https://docs.rapids.ai/api/cudf/stable/api_docs/dataframe.html lists a bunch of methods that are not yet supported, e.g., itertuples, iterrows, so the documentation seems a bit aspirational. I basically have a very large df, where I need to calculate the correlating matrix and then iterate over it. Interestingly, I also found some discussion online that mentions iterative calculations like iterrows are much faster on a CPU, so it doesn't really make sense to implement on a GPU. I think I might try the iterative part on a CPU again but would be curious to know if there're benefits to a GPU implementation, especially since it's part of the documentation already

**Comment by shwina** [comment]
Thanks for pointing this out, @sokol11 -- you're right that we incorrectly include `itertuples` and `iterrows` (and possibly a handful of other methods) as part of our API docs when in reality, we don't support those. I think the problem is that we _do_ have those methods defined, but all they do is raise a `NotImplementedError`. FYI @galipremsagar: we need to tackle this as part of our doc update effort

**Comment by vyasr** [comment]
Also @sokol11 to address this point of yours:
> Interestingly, I also found some discussion online that mentions iterative calculations like iterrows are much faster on a CPU, so it doesn't really make sense to implement on a GPU. 

This is exactly correct, and it is _why_ we define those methods are raising a `NotImplementedError`. We want to provide users a clear indication that working with cudf objects in this way is bad for performance. It's worth noting that even when using pandas there are usually much more efficient ways to get the answer you want. Here's an (admittedly trivial) example with just pandas

```
In [16]: import pandas as pd

In [17]: import numpy as np

In [18]: df = pd.DataFrame({'a': np.random.rand(10000)})

In [19]: %%timeit
    ...: total = 0
    ...: for idx, values in df.iterrows():
    ...:     total += values['a']
    ...: 
202 ms ± 760 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [20]: %%timeit
    ...: df['a'].sum()
    ...: 
    ...: 
34.6 µs ± 122 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

If you are using cudf for performance and you find yourself reaching for methods like `iterrows`, it's usually worth reconsidering your approach to see if there's a way to avoid iteration altogether.
